### PR TITLE
Add info about disabling IPv6 bsc#1152810

### PR DESCRIPTION
**Pull request by lcavajani**
Here are the info on how to disable IPv6 without encouraging it...

@nkoranova @r0ckarong there is an issue with `cilium.disable_ipv6` reference, do you have an idea how to fix this ?

```
Building DC-caasp-admin
daps -d /daps_temp-98g5jg6V02/source/DC-caasp-admin html  
daps -d /daps_temp-98g5jg6V02/source/DC-caasp-admin pdf  
Building DC-caasp-architecture
daps -d /daps_temp-98g5jg6V02/source/DC-caasp-architecture html  
daps -d /daps_temp-98g5jg6V02/source/DC-caasp-architecture pdf  
Building DC-caasp-deployment
daps -d /daps_temp-98g5jg6V02/source/DC-caasp-deployment html  
daps -d /daps_temp-98g5jg6V02/source/DC-caasp-deployment pdf  
Building DC-caasp-quickstart
/daps_temp-98g5jg6V02/source/build/.profiled/noprofile/caasp-quickstart.xml:367:89: error: IDREF "cilium.disable_ipv6" without matching ID
make: *** [validate] Error 1
/usr/share/daps/make/validate.mk:40: recipe for target 'validate' failed
Error: exit status 2
(Exiting d2d_runner) DC-caasp-quickstart has validation issues and cannot be built.
(Exiting d2d) Oh no. There are no documents for you.
```

https://github.com/SUSE/avant-garde/issues/920

Fixes bsc#1152810

Signed-off-by: lcavajani <lcavajani@suse.com>

### CODE REVIEW

**Comment by lcavajani** [comment]
cc @ereslibre

**Comment by innobead** [comment]
@ereslibre you mean when executing addon update it will cause customized manifests overrided by the current implementation?

**Comment by lcavajani** [comment]
@innobead The customization of the manifests is part of the whole addon upgrade but it is still not created. So maybe it's time to prioritized :)

(I assume in this doc that until we ship `1.16` the addon upgrade with kustomize would be ready)

**Comment by innobead** [comment]
> @innobead The customization of the manifests is part of the whole addon upgrade but it is still not created. So maybe it's time to prioritized :)
> 
> (I assume in this doc that until we ship `1.16` the addon upgrade with kustomize would be ready)

@lcavajani thanks for the info and yes, it looks high priority to me.

@jordimassaguerpla as u have an option to plan to release from master, please follow up this (customized manifest keep) to avoid overriding the customers deployment by 1.0.2 until it gets implemented. Thanks.

File: adoc/deployment-sysreqs.adoc [hunk 1]
```diff
@@ -203,13 +203,39 @@ The management workstation needs at least the following networking permissions:
...
 
-==== IP Addresses
+==== IPv6
+
+[WARNING]
+====
+Using IPv6 addresses is currently not supported but it is not disabled.
+====
+
+A common practice to disable IPv6 at the node level is to configure the following kernel parameter:
```
Comment (kkaempf) [inline adoc/deployment-sysreqs.adoc]: Where does this need to be configured ? Mention the exact file that needs to be adapted.

Comment (lcavajani) [reply adoc/deployment-sysreqs.adoc]: I didn't want to write a guide on how to disable IPv6 so I just refer to common practices used by sysadmins. IMO we should not encourage disabling IPv6.

Comment (lcavajani) [reply adoc/deployment-sysreqs.adoc]: I mean disabling it at the host level*

File: adoc/deployment-cilium.adoc [hunk 2]
```diff
@@ -40,3 +40,40 @@ https://cilium.readthedocs.io/en/v1.5/gettingstarted/memcached/
...
 Making security policies on clusters on AWS::
 https://cilium.readthedocs.io/en/v1.5/gettingstarted/aws/
+
+[[cilium-disable-ipv6]]
+==== Disable IPv6
+
+. Switch to the cluster file directory (e.g `my-cluster`).
+
+. In the cilium file `./addons/cni/cilium.yaml`, edit the args for the container
+with the name `cilium-agent` within the DaemonSet `cilium` to add "--enable-ipv6=false".
```
[the remark this answers is not in the document]
Comment (lcavajani) [reply adoc/deployment-cilium.adoc]: We could just disable it for every cluster but not sure this is the right path IMHO.

File: adoc/deployment-sysreqs.adoc (hunk 1 above)
Comment (kkaempf) [reply adoc/deployment-sysreqs.adoc]: > IMO we should not encourage disabling IPv6.

Well, customers do disable IPv6 for "security reasons" ;-) and don't care about our recommendations :-/

**Comment by ereslibre** [comment]
> @ereslibre you mean when executing addon update it will cause customized manifests overrided by the current implementation?

Yes, exactly. If we evolve the addon configuration/version bump, we will blindly reapply the new configuration, overriding the customer customizations.

Using `kustomize` would allow us to evolve our base while keeping customer customizations on top of it. I will open a RFC really soon tackling this.

File: adoc/deployment-sysreqs.adoc (hunk 1 above)
Comment (Martin-Weiss) [reply adoc/deployment-sysreqs.adoc]: It is a best practice and recommended to turn of all protocols that are not required and not used.
(not only for our systems ;-)) 
That is the reason why at least consulting recommends to turn of IPv6. 

As we do not support IPv6 with CaaSP / K8S at the moment and AFAIK K8S does not support dual-stack - I believe the default should be not to "depend" on it. 

Once we have IPv6 support this should be changed or at least the decision needs to be done during the deployment with Skuba.

Comment (lcavajani) [reply adoc/deployment-sysreqs.adoc]: Dual stack is in alpha in `1.16`.

We can just turn this into "enabling IPv6" if we go with disabling by default in skuba. No strong opinions on this.

**Comment by jordimassaguerpla** [comment]
> > @innobead The customization of the manifests is part of the whole addon upgrade but it is still not created. So maybe it's time to prioritized :)
> > (I assume in this doc that until we ship `1.16` the addon upgrade with kustomize would be ready)
> 
> @lcavajani thanks for the info and yes, it looks high priority to me.
> 
> @jordimassaguerpla as u have an option to plan to release from master, please follow up this (customized manifest keep) to avoid overriding the customers deployment by 1.0.2 until it gets implemented. Thanks.

Sorry I don't understand. Can you elaborate a bit more?

File: adoc/deployment-sysreqs.adoc (hunk 1 above)
Comment (Martin-Weiss) [reply adoc/deployment-sysreqs.adoc]: Can´t we enhance the cilium pod to autodetect if ipv6 is available?
(sounds a bit like working around a missing feature in that pod ;-))

**Comment by lcavajani** [comment]
Closing as `IPv6` is now disabled by default.

**Pull request closed without merging**